spring-integration-http/src/test/java/org/springframework/integration/http/inbound/MultipartPassThroughHttpMessageConverterTests.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.http.inbound;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.Enumeration;
+
+import javax.servlet.ServletInputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.junit.Test;
+import org.mockito.Matchers;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.integration.http.converter.MultipartPassThroughHttpMessageConverter;
+import org.springframework.messaging.Message;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+
+/**
+ * @author Gary Russell
+ * @since 4.2
+ *
+ */
+public class MultipartPassThroughHttpMessageConverterTests {
+
+	@SuppressWarnings("unchecked")
+	@Test
+	public void testMultiPass() throws Exception {
+		HttpRequestHandlingMessagingGateway gw = new HttpRequestHandlingMessagingGateway(false);
+		gw.setMessageConverters(
+				Collections.<HttpMessageConverter<?>> singletonList(new MultipartPassThroughHttpMessageConverter()));
```

Comment:
Yes; thanks, `BAHMC` works fine too - it `canRead()` `MediaType.ALL`.

Changing...